web/docs/tql2/operators/deduplicate.md

```diff
@@ -0,0 +1,87 @@
+# deduplicate
+
+Removes duplicate events based on a common key.
+
+```tql
+deduplicate [key:any, limit=int, distance=int, create_timeout=duration,
+             write_timeout=duration, read_timeout=duration]
+```
+
+## Description
+
+The `deduplicate` operator removes duplicates from a stream of events, based
+on the value of one or more fields.
+
+The operator has four tuning knobs:
+
+1. **Key**: The expression to deduplicate. Defaults to `this`, i.e.,
+   deduplicating entire events. To deduplicate multiple fields, use a record
+   expression like `{foo: bar, baz: qux}`.
+2. **Limit**: The number of duplicate keys allowed before an event is
+   suppressed. Defaults to 1, which is equivalent to removing all duplicates.
+3. **Distance**: Distance between two events that can be considered duplicates.
+   A value of 1 means that only adjacent events can be considered duplicate.
+   When unspecified, the distance is infinite.
+4. **Timeout**: The time that needs to pass until a surpressed event is no
+   longer considered a duplicate. The create timeout resets when the first event
+   for a given key is let through, the write timeout resets when any event for a
+   given key is let through, and the read timeout is reset when a key is seen,
+   even if the event is suppressed.
```

Comment:
This uses a very different format for documenting the arguments than we usually do. Not saying it's a bad thing (I was also inclined towards more free form docs), but do we want to keep it that way?

(I would not block the PR for this, but the other conversation will anyway)

Reply:
I copied this from some other operator, and didn't think about it too deeply. Will align this.

Reply:
Done and pushed.